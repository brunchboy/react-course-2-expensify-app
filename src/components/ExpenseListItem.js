import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = props => (
  <div>
    <Link to={`/edit/${props.id}`}>
      <h2>{props.description}</h2>
    </Link>
    <p>
      {props.amount} - {moment(props.createdAt).format()}
    </p>
  </div>
);

export default ExpenseListItem;
