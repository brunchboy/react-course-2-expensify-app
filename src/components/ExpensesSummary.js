import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  const pluralized = expensesCount == 1 ? 'expense' : 'expenses';
  const formattedTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <h2>
      Viewing {expensesCount} {pluralized} totalling {formattedTotal}
    </h2>
  );
};

const mapStateToProps = state => {
  const expenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: expenses.length,
    expensesTotal: getExpensesTotal(expenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
