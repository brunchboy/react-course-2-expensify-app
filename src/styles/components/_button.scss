.button {
  border: 0;
  border-radius: 0px;
  box-shadow: none;
  background: $dark-cyan;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 350;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888888;
}
