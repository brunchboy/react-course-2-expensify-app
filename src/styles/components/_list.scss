.list-header {
  background: $vellum;
  border: 1px solid darken($vellum, 7%);
  color: gray;
  display: flex;
  justify-content: space-between;
  padding: $s-size $m-size;
}

.list-body {
  margin-bottom: $m-size;
  @media (min-width: $desktop-breakpoint) {
    margin-bottom: $l-size;
  }
}

.list-item {
  border: 1px solid darken($vellum, 7%);
  border-top: none;
  color: $dark-gray;
  display: flex;
  flex-direction: column;
  padding: $s-size;
  text-decoration: none;
  transition: background 0.3s ease;
  &:hover {
    background: $vellum;
  }
  @media (min-width: $desktop-breakpoint) {
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
  }
}

.list-item--message {
  align-items: center;
  color: $gray;
  justify-content: center;
  padding: $m-size;
  &:hover {
    background: none;
  }
}

.list-item__title {
  margin: 0;
  word-break: break-all;
}

.list-item__subtitle {
  color: $gray;
  font-size: $font-size-small;
}

.list_item__data {
  margin: $s-size 0 0 0;
  @media (min-width: $desktop-breakpoint) {
    margin: 0;
    padding-left: $s-size;
  }
}
