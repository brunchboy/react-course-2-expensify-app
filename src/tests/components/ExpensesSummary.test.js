import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render correct summary with all expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expensesCount={3} expensesTotal={114195} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render correct summary with a single expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expensesCount={1} expensesTotal={4500} />
  );
  expect(wrapper).toMatchSnapshot();
});
