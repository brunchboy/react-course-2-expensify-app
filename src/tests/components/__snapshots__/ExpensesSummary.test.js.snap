// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct summary with a single expense 1`] = `
<h2>
  Viewing 
  1
   
  expense
   totalling 
  $45.00
</h2>
`;

exports[`should render correct summary with all expenses 1`] = `
<h2>
  Viewing 
  3
   
  expenses
   totalling 
  $1,141.95
</h2>
`;
